proto/prehog/v1/teleport.proto

```diff
@@ -124,6 +124,10 @@ message UserActivityRecord {
   uint64 certificates_issued = 17;
   // counter of SVIDs issued for each SPIFFE ID.
   repeated SPIFFEIDRecord spiffe_ids_issued = 18;
+  // TODO(sshah): remove reserve comment below before merging to master.
+  // 19,20,21 reserved for user_origin, access_request_created, access_request_reviewed, access_list_grants
+  // counter of saml.idp.auth events
+  uint64 saml_idp_sessions = 24;
```

Comment:
Why not just add a field with the next available proto field number and extend it in order,  instead of reserving it for user_origin, access_request_created, access_request_reviewed, and access_list_grants before even these filed are added ? 

```suggestion
  uint64 saml_idp_sessions = 19;
```

I feel like reversing the proto field number for a field that will be added in the future is bad practice.

Reply:
Pardon the confusion. This is reserving fields added in the https://github.com/gravitational/teleport/pull/52183 and https://github.com/gravitational/teleport/pull/52170 PR, that came in draft before this PR but I ended opening this one for review before them. 

Anyway, this is not reserving the actual proto number per se, if you look at the comment. While working with the prehog rpc, all these PRs need to pair with the cloud prehog PR sibling for testing in the dev Posthog instance. So it helps to keep them the same. And these should not be merged until the Cloud sibling PR is merged to master.

Reply:
Still, do not change the order of proto fields, but add uint64 saml_idp_sessions = 19;.

We don’t have a guarantee that a new field won’t be added while working in the monorepo.

So, in terms of proto field numbering, assigning int64 saml_idp_sessions = 24; doesn’t make sense. If you remove the comment:

> // TODO(sshah): remove reserve comment below before merging to master.
> // 19,20,21 reserved for user_origin, access_request_created, access_request_reviewed, access_list_grants
> // counter of saml.idp.auth events


without aligning the proto number (which would be a breaking change), there will be a gap in proto numbering between 19 and 24 and this is something that we should avoid.

Reply:
> We don’t have a guarantee 

In this case, we need to guarantee that the proto numbering in Teleport OSS matches with the cloud  prehog proto. Otherwise the cloud prehog instance will be shipped with different proto version and will never work.

Reply:
> Does the cloud prehog sibling PR was already merged ?




This is way we are requiring that the cloud change is merged first: 

> When changing the definitions, please ensure that the resulting proto files are merged in cloud's master branch before merging in Teleport's master or release branches.

And become and source of true the teleport should mirror the def. 

In current approach the teleport proto is aligned without cloud change. 

In the current `ecord saml_idp_sessions event` approach, the Teleport proto is aligned without changes to the cloud.

For compatibility, this PR should be merged after the cloud proto definitions are aligned and merged.